pkg/slip0044/slip0044.go

```diff
@@ -0,0 +1,23 @@
+package slip0044
+
+import "strings"
+
+// Return the path component that corresponds to this coin type.
+func (t CoinType) PathComponent() uint32 {
+	return 0x80000000 | uint32(t)
+}
+
+// Parse a coin from its symbol or name into the corresponding coin type.
+func ParseCoinType(coin string) (CoinType, error) {
+	// First try to parse as a symbol.
+	if coinType, ok := typesBySymbol[strings.ToUpper(coin)]; ok {
+		return coinType, nil
+	}
+
+	// Otherwise try to parse as a name.
+	if coinType, ok := typesByName[strings.ToLower(coin)]; ok {
+		return coinType, nil
+	}
+
+	return 0, ErrUnknownCoin
+}
```

Comment:
Do we need to have the reverse mappings here as well? e.g. the coin type number to the name or symbol?

Reply:
Possibly, I think most people think about asset types in terms of the symbol -- so that is the primary reference type. But we can cross that bridge when we get to it - e.g. when it comes to JSON serialization.